test/functional/normalization/eventrouter_test.go

```diff
@@ -0,0 +1,105 @@
+package normalization
+
+import (
+	"encoding/json"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/pkg/utils"
+	"github.com/openshift/cluster-logging-operator/test/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/reference"
+)
+
+var _ = Describe("[Normalization] Fluentd normalization for EventRouter messages", func() {
+
+	var (
+		framework *functional.FluentdFunctionalFramework
+		pod *corev1.Pod
+		podName, ns string
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewFluentdFunctionalFramework()
+		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+			FromInput(logging.InputNameApplication).
+			ToFluentForwardOutput()
+		Expect(framework.Deploy()).To(BeNil())
+		pod = types.MockPod()
+		podName = pod.Name
+		ns = pod.Namespace
+	})
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	It("Should parse eventrouter `ADDED` message and check values", func() {
+		podRef, err := reference.GetReference(scheme.Scheme, pod)
+		Expect(err).To(BeNil())
+		reason := string(utils.GetRandomWord(8))
+		message := string(utils.GetRandomWord(32))
+		newEvent := types.MockEvent(podRef, corev1.EventTypeNormal, reason, message)
+		newEventData := types.EventData{Verb: "ADDED", Event: newEvent}
+		Expect(framework.WritesEventrouterLogs([]types.EventData{newEventData})).To(BeNil())
+		raw, err := framework.ReadEventRouterLogsFrom("fluentforward")
+
+		Expect(err).To(BeNil(), "Expected no errors reading the logs")
+		logs, err := types.ParseLogs(raw)
+		Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+		eventData := types.EventData{}
+		Expect(logs[0].Message).ToNot(BeNil(), fmt.Sprintf("Expected to find the eventrouter message  %#v", logs[0]))
+		err = json.Unmarshal([]byte(logs[0].Message), &eventData)
+		Expect(err).To(BeNil())
+		Expect(eventData).ToNot(BeNil(), "Expected to find the EventData")
+		Expect(eventData.Verb).To(BeEquivalentTo("ADDED"))
+		event := eventData.Event
+		Expect(event).ToNot(BeNil())
+		Expect(event.Type).To(BeEquivalentTo(corev1.EventTypeNormal))
+		Expect(event.Message).To(BeEquivalentTo(message))
+		Expect(event.Reason).To(BeEquivalentTo(reason))
+		Expect(event.InvolvedObject.Name).To(BeEquivalentTo(podName))
+		Expect(event.InvolvedObject.Namespace).To(BeEquivalentTo(ns))
+	})
+
+	It("Should parse eventrouter `UPDATED` message and check values", func() {
```

Comment:
@alanconway @eranra What are your thoughts on writing this test using the template style that @eranra introduced for audit logs?  Is there reason we should be validating values more strictly or is it sufficient to know we transformed an "event" log to include the desired fields?

Reply:
I think it is sufficient to check the expected change, though I wouldn't object to more validation.
As long as we have some tests that do full validation, I don't think every test needs to - unless testing something that is likely to invalidate unknown parts of the record.

Reply:
@jcantrill the purpose of that templates idea was exactly cases like this one that there are multiple `expect` statements and instead of that code using one nice and more understandable template. I get @alanconway point that it is not critical ... I can help to create if this is needed,.

BTW; It is consistent in the code (tests, also prod) that there are multiple ways/frameworks for same functionalities ... I see that as very sick area of the code base and that consolidating that into more agreed and coherent one-way of doing things will accelerate development and increase correctness and debuggability. So for example in the templates case ... I advice either to start using that in all new tests (like this one) or revert to multiple `Expect` statements ... having both for long time in the code without pure direction does not feel good to me.

Reply:
I can rewrite with `FitLogFormatTemplate` function, but I think it will be complicated for code reading because need to know how actually function works. BTW `FitLogFormatTemplate` also can provide errors. But I agree that need to do in one way

Reply:
@vparfonov please rewrite the test to use the template validation so we have consistency

Reply:
ok